src/DeploymentProvider/DeploymentHelper.ts

```diff
@@ -100,29 +100,97 @@ export class DeploymentHelper {
+        const buildResultName = regex.exec(buildResponse.properties.triggeredBuildResult.id)[0];
+        let buildProvisioningState = 'Queuing';
+        let cnt = 0;
+        core.debug("wait for build result......");
+        //Waiting for build result. Timeout 30 minutes.
+        while (buildProvisioningState != 'Succeeded' && buildProvisioningState != 'Failed' && cnt++ < 180) {
+            await new Promise(resolve => setTimeout(resolve, 10000));
+            console.log("wait for 10 seconds....");
+            const waitResponse = await client.buildServiceOperations.getBuildResult(params.resourceGroupName, params.serviceName, buildServiceName, buildName, buildResultName);
+            core.debug('build result response: ' + JSON.stringify(waitResponse));
+            buildProvisioningState = waitResponse.properties.provisioningState;
+        }
+        if (cnt == 180) {
+            throw Error("Build result timeout.");
+        }
+        if (buildProvisioningState != 'Succeeded') {
+            throw Error("Build result failed.");
+        }
+        let deploymentResource: asa.DeploymentResource = await this.buildDeploymentResource(client, params, sourceType, buildResponse.properties.triggeredBuildResult.id);
+        core.debug("deploymentResource: " + JSON.stringify(deploymentResource));
+        const deployResponse = await client.deployments.beginCreateOrUpdateAndWait(params.resourceGroupName, params.serviceName, params.appName, params.deploymentName, deploymentResource);
+        core.debug('deploy response: ' + JSON.stringify(deployResponse));
+
+    }
+
+    public static async deleteDeployment(client: asa.AppPlatformManagementClient, params: ActionParameters) {
+        const response = await client.deployments.beginDeleteAndWait(params.resourceGroupName, params.serviceName, params.appName, params.deploymentName);
+        core.debug('delete deployment response: ' + JSON.stringify(response));
+        return;
+    }
+
+    protected static async buildDeploymentResource(client: asa.AppPlatformManagementClient, params: ActionParameters, sourceType: string, idOrPath: string): Promise<asa.DeploymentResource> {
         let getDeploymentName = params.deploymentName;
         if (params.createNewDeployment) {
             getDeploymentName = await this.getProductionDeploymentName(client, params);
         }
         let getResponse: asa.DeploymentResource = await this.getDeployment(client, params, getDeploymentName);
         let deploymentResource: asa.DeploymentResource;
-        let sourcePart: asa.UploadedUserSourceInfoUnion = {
-            relativePath: uploadResponse.relativePath,
-            type: sourceType as "UploadedUserSourceInfo" | "Jar" | "Source" | "NetCoreZip"
-        };
+        let sourcePart = {};
+        sourcePart["type"] = sourceType;
+        const idOrPathKey = sourceType == "BuildResult" ? "buildResultId" : "relativePath";
+        sourcePart[idOrPathKey] = idOrPathKey;
```

Comment:
Sorry I made a mistake here. We also need to switch on `buildResultId` and `relativePath`. The original implementation would be more clear in this case.

However, we can use `SourceType.BUILD_RESULT` to avoid the magic string `BuildResult`